libs/motor_control/esc/esc.h

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of the efc project <https://github.com/eurus-project/efc/>.
+ * Copyright (c) 2024-2025 The efc developers.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef ESC_H
+#define ESC_H
+
+#include <stdio.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/kernel.h>
+
+#define ESC_APPLY_MIN_OFFSET(OFFSET, DC_MIN, DC_MAX)                           \
+    ((DC_MIN) + (((DC_MAX) - (DC_MIN)) * ((OFFSET) / 100.0f)))
+#define ESC_APPLY_MAX_OFFSET(OFFSET, DC_MIN, DC_MAX)                           \
+    ((DC_MAX) - (((DC_MAX) - (DC_MIN)) * ((OFFSET) / 100.0f)))
+
+// Defined in Kconfig
+#if CONFIG_ESC_MANUAL_MIN_THRUST_OFFSET
```

Comment:
This is defined in the demo Kconfig but you're checking for it in the library, this won't exist when the library is used outside of the demo and also it is very unintuitive to depend on app Kconfig for a library, if configuring a library create a Kconfig for that library.